tools/swift-format/swift-format.cpp

```diff
@@ -0,0 +1,350 @@
+      }
+
+      const MemoryBuffer &memBuffer() const {
+        return *SM.getLLVMSourceMgr().getMemoryBuffer(BufferID);
+      }
+
+      std::pair<unsigned, unsigned>
+      getLineAndColumn(unsigned ByteOffset) const {
+        auto &Buf = memBuffer();
+        if (ByteOffset > Buf.getBufferSize())
+          return std::make_pair(0, 0);
+
+        SMLoc Loc = SMLoc::getFromPointer(Buf.getBufferStart() + ByteOffset);
+        return SM.getLLVMSourceMgr().getLineAndColumn(Loc, BufferID);
+      }
+
+    };
+
+    // Parses <start line>:<end line> input to a pair of line numbers.
+    // Returns true on error.
+    static bool parseLineRange(StringRef Input, unsigned &FromLine,
+                               unsigned &ToLine) {
+      std::pair<StringRef, StringRef> LineRange = Input.split(':');
+      return LineRange.first.getAsInteger(0, FromLine) || LineRange.second.getAsInteger(0, ToLine);
+    }
+
+    static void outputReplacementXML(StringRef Text) {
+      // FIXME: When we sort includes, we need to make sure the stream is correct
+      // utf-8.
+      size_t From = 0;
+      size_t Index;
+      while ((Index = Text.find_first_of("\n\r<&", From)) != StringRef::npos) {
+        llvm::outs() << Text.substr(From, Index - From);
+        switch (Text[Index]) {
+        case '\n':
+          llvm::outs() << "&#10;";
+          break;
+        case '\r':
+          llvm::outs() << "&#13;";
+          break;
+        case '<':
+          llvm::outs() << "&lt;";
+          break;
+        case '&':
+          llvm::outs() << "&amp;";
+          break;
+        default:
+          llvm_unreachable("Unexpected character encountered!");
+        }
+        From = Index + 1;
+      }
+      llvm::outs() << Text.substr(From);
+    }
+
+    static void outputReplacementsXML(const Replacements &Replaces) {
+      for (const auto &R : Replaces) {
+        outs() << "<replacement "
+               << "offset='" << R.getOffset() << "' "
+               << "length='" << R.getLength() << "'>";
+        outputReplacementXML(R.getReplacementText());
+        outs() << "</replacement>\n";
+      }
+    }
+
+    // Returns true on error.
+    static bool format(StringRef FileName) {
+      ErrorOr<std::unique_ptr<MemoryBuffer>> CodeOrErr =
+        MemoryBuffer::getFileOrSTDIN(FileName);
+      if (std::error_code EC = CodeOrErr.getError()) {
+        llvm::errs() << EC.message() << "\n";
+        return true;
+      }
+      std::unique_ptr<llvm::MemoryBuffer> Code = std::move(CodeOrErr.get());
+      if (Code->getBufferSize() == 0)
+        return false; // Empty files are formatted correctly.
+
+      FormatterDocument Doc(std::move(Code));
+
+      if (!Offsets.empty() || !Lengths.empty()) {
+        if (Offsets.size() != Lengths.size()) {
+          llvm::errs() << "error: number of offsets not equal to number of lengths.\n";
+          return true;
+        }
+
+        for ( unsigned i=0 ; i < Offsets.size() ; i++ ) {
```

Comment:
`for (unsigned i = 0; i < Offsets.size(); i++) {` is the dominant spacing style used in the repo :-)

I don't normally comment on coding style but since this is a coding style PR I'm making an exception :-)